crud.rb

```diff
@@ -46,19 +61,42 @@ def initialize
   erb :"users/edit"
 end
 
+# Login - Checks user credentials
+#-----------------------------------------------------------------
+post '/users/login' do
+  @user = @users.where(username: params[:username]).first
+  if @user == nil
+    erb :"homepage/error"
+  else
+    # password_salt = @user[:password_salt]
+    password = BCrypt::Engine.hash_secret(params[:password], @user[:password_salt])
+    if @user[:password_hash] == password
+      erb :"users/show"
```

Comment:
Instead of this, use `redirect "/users/#{id}"` to redirect the user to the show page. It's a good practice to make all non-`GET` methods to always redirect. Only `GET` methods should display pages.